test/request/request.go

```diff
@@ -1,28 +1,77 @@
 package request
 
 import (
-	"io"
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
 	"net/http"
+	"net/url"
 
 	"github.com/alibaba/pouch/client"
 	"github.com/alibaba/pouch/pkg/utils"
 	"github.com/alibaba/pouch/test/environment"
 )
 
+// Option defines a type used to update http.Request.
+type Option func(*http.Request) error
+
+// WithQuery sets the query field in URL.
+func WithQuery(query url.Values) Option {
+	return func(r *http.Request) error {
+		r.URL.RawQuery = query.Encode()
+		return nil
+	}
+}
+
+// WithJSONBody sets the body in http.Request
+func WithJSONBody(obj interface{}) Option {
+
+	return func(r *http.Request) error {
+		b := bytes.NewBuffer([]byte{})
+
+		if obj != nil {
+
+			err := json.NewEncoder(b).Encode(obj)
+
+			if err != nil {
+				return err
+			}
+		}
+		r.Body = ioutil.NopCloser(b)
```

Comment:
I think we can add `r.Header.Set("Content", "application-json")` to make it better. It's not required, but it can make request more clear.

Reply:
good~

Reply:
I think add a WithHeader function to do this is better